### PR TITLE
chore: rework DataSelector styles to css modules

### PR DESCRIPTION
**Pull request by uladzimirdev**
[epic](https://github.com/metabase/metabase/issues/50983)

### Description

PR 3/?

Rework DataSelector to css modules

### How to verify
verify pickers. e.g. follow steps from this [PR](https://github.com/metabase/metabase/pull/50535)

Screenshot 2024-12-05 at 17 18 18

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelector.module.css [hunk 1]
```diff
@@ -0,0 +1,45 @@
+.TableSearchContainer {
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/DataSelector/DataSelector.module.css:1]: I'm pretty sure our convention is to use `.camelCase`

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelectorSchemaPicker/DataSelectorSchemaPicker.tsx [hunk 2]
```diff
@@ -36,7 +34,7 @@ const DataSelectorSchemaPicker = ({
   ];
 
   return (
-    <Container>
+    <Box w={300}>
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/DataSelector/DataSelectorSchemaPicker/DataSelectorSchemaPicker.tsx:37]: This `300` comes up a lot in this PR. Maybe we should have a constant for it, WDYT?

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx [hunk 3]
```diff
@@ -123,6 +121,18 @@ export function FieldDataSelector(props) {
   );
 }
 
+export function RawDataBackButton() {
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx:124]: nit: It's not used here, can we move it to a a separate file?

**Review by kamilmielnik — APPROVED** [review]
Works like it used to :+1:

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelector.module.css (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/DataSelector/DataSelector.module.css:1]: right, but I wanted to have one to one migration from StyledComponents to css modules so it's easier to review when the class name is the same as a component name was

Comment (kamilmielnik) [reply frontend/src/metabase/query_builder/components/DataSelector/DataSelector.module.css:1]: That's cool but I think it was not necessary :)

**Comment by uladzimirdev** [comment]
Fixed review comments except class names, I believe some automation can help later with renaming all class names and their usages at once.

Also added 2 more css modules about saved entity picker after review from Kamil, but they are straightforward and shouldn't break anything

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx]: done

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelectorSchemaPicker/DataSelectorSchemaPicker.tsx (hunk 2 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/DataSelector/DataSelectorSchemaPicker/DataSelectorSchemaPicker.tsx]: added a constant and reused it in DataSelector folder

**Pull request merged**